app/controllers/api/irs_attempts_api_controller.rb

```diff
@@ -24,8 +24,11 @@ def create
         headers['X-Payload-Key'] = Base64.strict_encode64(result.encrypted_key)
         headers['X-Payload-IV'] = Base64.strict_encode64(result.iv)
 
-        send_data result.encrypted_data,
-                  disposition: "filename=#{result.filename}"
+        bucket_name = IdentityConfig.store.irs_attempt_api_bucket_name
+
+        requested_data = @s3_client.get_object(bucket: bucket_name, key: result.filename)
+
+        send_data requested_data, disposition: "filename=#{result.filename}"
```

Comment:
could of quick notes:
1. I think we need a feature flag for if this endpoint reads from S3 or dynamically builds the responses
2. I think when the feature flag says read from s3, we should either be querying S3 based on the timestamp, or checking our DB logs to see if we have the file, before trying to read from s3
3. I think we need a `.body` when sending data like this
```suggestion
        bucket_name = IdentityConfig.store.irs_attempt_api_bucket_name

        requested_data = @s3_client.get_object(bucket: bucket_name, key: result.filename)

        send_data requested_data.body, disposition: "filename=#{result.filename}"
```

Reply:
Thanks for the great feedback!  I believe I have incorporated all of your suggestions.  Please comment as needed!